### ISSUE TITLE
Packaging github.com/gookit/color [#1052019]

### ISSUE THREAD

**Issue by jmkim** [jesseduffield/lazygit-debian#33]
https://salsa.debian.org/go-team/packages/golang-github-gookit-color

**Comment by afeedhshaji** [comment]
Hi, I'm new to Debian packaging but would love to try it out on this go library. 
I'm assuming the steps here are to:
- [ ] Create the debian package files, etc. for this repo. I'm not sure where to place this repo though. (Salsa git?)
- [ ]  Send an ITA and RFS

**Comment by jmkim** [comment]
> Create the debian package files, etc. for this repo. I'm not sure where to place this repo though. (Salsa git?)

Normally it should be in `https://salsa.debian.org/go-team/packages/PKG_NAME`
The repository in the team's namespace can be created by a command: `dh-make-golang create-salsa-project PKG_NAME`.

Note that if you are not DD, you CANNOT force-push, revert, remove branch, remove tag, or any kinds of setting about the repository.

> Send an ITA and RFS

Suppose ITP and RFS you meant.

ITP could be filed with the command `reportbug` to the pseudo-package `wnpp`. You can find the [ITP template for lazygit is here](https://github.com/jesseduffield/lazygit-debian/wiki/Go-ITP-template). Because of lazygit and its dependencies is team-maintained (Go Packaging team), please don't forget the `X-Debbugs-CC: debian-go@lists.debian.org`.

RFS could be sent mail directly to team mailing lists [`debian-go@lists.debian.org`](debian-go@lists.debian.org).  You can find the [RFS template for lazygit is here](https://github.com/jesseduffield/lazygit-debian/wiki/Go-RFS-template).

**Comment by jmkim** [comment]
I read the ITP mail from you @afeedhshaji, so I assigned you here \o/

ITP: https://bugs.debian.org/1052019

**Comment by afeedhshaji** [comment]
I have created the Salsa repo in the go-team org: https://salsa.debian.org/go-team/packages/golang-github-gookit-color. However I'm unable to push code to this repository. How do I get the permission to do it?

**Comment by jmkim** [comment]
@afeedhshaji, just after DebConf23, now I arrived Korea \o/

Did you join the [go-team salsa](https://salsa.debian.org/go-team)? If not, you have to join here first.

 1. Click "Request Access" button, which is in [go-team salsa main page](https://salsa.debian.org/go-team).
 2. Mail to `debian-go@lists.debian.org`.
    * Example mail : https://lists.debian.org/debian-go/2023/09/msg00022.html

**Comment by jmkim** [comment]
Note that, if you are not a [DD](https://wiki.debian.org/DebianDeveloper), you **CANNOT** `force-push, revert, remove branch, remove tag, or any kinds of repo settings` regarding to the repository.

As I am not DD, I cannot help those. For any of them, we have to ask `debian-go@lists.debian.org`.

Push very carefully \o/

**Comment by afeedhshaji** [comment]
Thanks @jmkim. I did request for access in the repo and did not get access on till now. Guess I needed to send the mail! :sweat:

**Comment by afeedhshaji** [comment]
>..just after DebConf23, now I arrived Korea \o/

No worries. Hope you had a great time!

Thanks @jmkim. I did request for access in the repo and did not get access on it till now. Guess I needed to send the mail! 😓

**Comment by afeedhshaji** [comment]
Hello @jmkim !  I have pushed the code to https://salsa.debian.org/go-team/packages/golang-github-gookit-color. I had a few queries regarding the pkg. 
1. I had to disable tests since although the tests were running fine, the test runner seems to be exiting with a non-zero exit code.
2. The `.deb` file was created and installed. But I'm not quite sure how folks normally "test" out go libraries in the Debian packagin world. I tested it out by copying the `/usr/share/gocode/src/github.com/gookit/color` (which I got after deb installing) to the `GOPATH` pkg `~/go/pkg/mod/github.com/gookit` and ran it with a sample go program which seems to be working.

**Comment by jmkim** [comment]
### QA1
> 1. I had to disable tests since although the tests were running fine, the test runner seems to be exiting with a non-zero exit code.

```
=== RUN   TestUtilFuncs
    utils_test.go:21: 
        	Error Trace:	/build/golang-github-gookit-color-1.5.4/_build/src/github.com/gookit/color/utils_test.go:21
        	Error:      	Received unexpected error:
        	            	open README.md: no such file or directory
        	Test:       	TestUtilFuncs
--- FAIL: TestUtilFuncs (0.00s)
```

One error occurred, which seems need `README.md` as a fixture.

I believe disable only this test (`TestUtilFuncs`) is better than entire test.
* You can refer [how I did before for another package](https://salsa.debian.org/go-team/packages/golang-github-jesseduffield-go-getter/-/blob/debian/sid/debian/patches/disable-internet-access.patch).
* Comment out the specific test and [make it as a Debian patch](https://wiki.debian.org/UsingQuilt).

### QA2

> 2. The .deb file was created and installed. But I'm not quite sure how folks normally "test" out go libraries in the Debian packagin world. I tested it out by copying the /usr/share/gocode/src/github.com/gookit/color (which I got after deb installing) to the GOPATH pkg ~/go/pkg/mod/github.com/gookit and ran it with a sample go program which seems to be working.

In Debian, Go package will be built with `dh-golang`. So when we want to dig, [we can refer the `dh-golang` code](https://salsa.debian.org/go-team/packages/dh-golang/-/blob/debian/sid/lib/Debian/Debhelper/Buildsystem/golang.pm).
* [`dh-golang` manpage (parameter guides)](https://manpages.debian.org/testing/dh-golang/Debian::Debhelper::Buildsystem::golang.3pm.en.html)
* [`dh-golang` code (implementation)](https://salsa.debian.org/go-team/packages/dh-golang/-/blob/debian/sid/lib/Debian/Debhelper/Buildsystem/golang.pm)

For the case above (`README.md` file thing), we can refer:
* [List of file extension when Debian Go package built and installed](https://salsa.debian.org/go-team/packages/dh-golang/-/blob/debian/sid/lib/Debian/Debhelper/Buildsystem/golang.pm#L521)
* [Why go.mod and go.sum are installed too](https://salsa.debian.org/go-team/packages/dh-golang/-/blob/debian/sid/lib/Debian/Debhelper/Buildsystem/golang.pm#L556)

These do not have `.md` as an installed extension, so `README.md` will not be installed into `/usr/share/gocode/src/`.


If we want to force-install `README.md`, we can pass it into `DH_GOLANG_INSTALL_EXTRA`.
* [`DH_GOLANG_INSTALL_EXTRA` in `dh-golang` manpage](https://manpages.debian.org/testing/dh-golang/Debian::Debhelper::Buildsystem::golang.3pm.en.html#DH_GOLANG_INSTALL_EXTRA)
* [`DH_GOLANG_INSTALL_EXTRA` in `dh-golang` code](https://salsa.debian.org/go-team/packages/dh-golang/-/blob/debian/sid/lib/Debian/Debhelper/Buildsystem/golang.pm#L571)
* You can refer [how I did before for another package](https://salsa.debian.org/go-team/packages/golang-github-fsmiamoto-git-todo-parser/-/blob/debian/sid/debian/rules#L5)

But I think this is not make sense, because the file will only used as test fixture. I agree to disable the test, as I commented in QA1.

**Comment by jmkim** [comment]
**TL;DR:** [In QA1;](https://github.com/jesseduffield/lazygit-debian/issues/33#issuecomment-1742643664) I believe disable only this test (`TestUtilFuncs`) is better than entire test. \o/

**Comment by afeedhshaji** [comment]
Thanks for taking your time in addressing my queries, Jongmin! I have sent the RFS here: https://lists.debian.org/debian-go/2023/10/msg00005.html

**Comment by jmkim** [comment]
Now I am wrapping up, and I can see [the package was uploaded](https://tracker.debian.org/pkg/golang-github-gookit-color). Thanks for your contribution \o/
